pkg/chunk/grpc/grpc_server_mock.go

```diff
@@ -0,0 +1,203 @@
+func (s server) CreateTable(ctx context.Context, desc *TableDesc) (*empty.Empty, error) {
+	if desc.Name == "chunk_2607" && !desc.UseOnDemandIOMode && desc.ProvisionedRead == 300 && desc.ProvisionedWrite == 1 && desc.Tags == nil {
+		return &empty.Empty{}, nil
+	}
+	err := errors.New("create table from tableClient request doesn't match with test gRPC client")
+	return &empty.Empty{}, err
+}
+
+func (s server) DeleteTable(ctx context.Context, name *TableName) (*empty.Empty, error) {
+	if name.TableName == "chunk_2591" {
+		return &empty.Empty{}, nil
+	}
+	err := errors.New("delete table from tableClient request doesn't match with test gRPC client")
+	return &empty.Empty{}, err
+}
+
+func (s server) DescribeTable(ctx context.Context, name *TableName) (*DescribeTableResponse, error) {
+	if name.TableName == "chunk_2591" {
+		return &DescribeTableResponse{
+			Desc: &TableDesc{
+				Name:              "chunk_2591",
+				UseOnDemandIOMode: false,
+				ProvisionedRead:   0,
+				ProvisionedWrite:  0,
+				Tags:              nil,
+			},
+			IsActive: true,
+		}, nil
+	}
+	err := errors.New("describe table from tableClient request doesn't match with test gRPC client")
+	return &DescribeTableResponse{}, err
+}
+
+func (s server) UpdateTable(ctx context.Context, request *UpdateTableRequest) (*empty.Empty, error) {
+	if request.Current.Name == "chunk_2591" && !request.Current.UseOnDemandIOMode && request.Current.ProvisionedWrite == 0 &&
+		request.Current.ProvisionedRead == 0 && request.Current.Tags == nil && request.Expected.Name == "chunk_2591" &&
+		!request.Expected.UseOnDemandIOMode && request.Expected.ProvisionedWrite == 1 &&
+		request.Expected.ProvisionedRead == 300 && request.Expected.Tags == nil {
+		return &empty.Empty{}, nil
+	}
+	err := errors.New("update table from tableClient request doesn't match with test gRPC client")
+	return &empty.Empty{}, err
+}
+
+func (s server) Stop(ctx context.Context, empty *empty.Empty) (*empty.Empty, error) {
+	return empty, nil
+}
+
+// NewStorageClient returns a new StorageClient.
+func NewTestStorageClient(cfg Config, schemaCfg chunk.SchemaConfig) (*StorageClient, error) {
+	grpcClient, _, err := connectToTestGrpcServer(cfg.Address)
+	if err != nil {
+		return nil, err
+	}
+	client := &StorageClient{
+		schemaCfg: schemaCfg,
+		client:    grpcClient,
+	}
+	return client, nil
+}
+
+//***********************  gRPC mock server *********************************//
+
+// NewTableClient returns a new TableClient.
+func NewTestTableClient(cfg Config) (*TableClient, error) {
+	grpcClient, _, err := connectToTestGrpcServer(cfg.Address)
+	if err != nil {
+		return nil, err
+	}
+	client := &TableClient{
+		client: grpcClient,
+	}
+	return client, nil
+}
+
+// NewStorageClient returns a new StorageClient.
+func newTestStorageServer(cfg Config) (*server, error) {
+	client := &server{
+		Cfg: cfg,
+	}
+	return client, nil
+}
+
+func connectToTestGrpcServer(serverAddress string) (GrpcStoreClient, *grpc.ClientConn, error) {
```

Comment:
How is this different than `connectToGrpcServer()`? Why not using `connectToGrpcServer()`?

Reply:
yeah. We can use it.